analyzers/src/SonarAnalyzer.CSharp/Rules/MagicNumberShouldNotBeUsed.cs

```diff
@@ -61,52 +61,49 @@ public sealed class MagicNumberShouldNotBeUsed : SonarDiagnosticAnalyzer
 
         private static bool IsExceptionToTheRule(LiteralExpressionSyntax literalExpression) =>
             NotConsideredAsMagicNumbers.Contains(literalExpression.Token.ValueText)
-            // It's ok to use magic numbers as part of a variable declaration
             || literalExpression.FirstAncestorOrSelf<VariableDeclarationSyntax>() != null
-            // It's ok to use magic numbers as part of a parameter declaration
             || literalExpression.FirstAncestorOrSelf<ParameterSyntax>() != null
-            // It's ok to use magic numbers as part of an enum declaration
             || literalExpression.FirstAncestorOrSelf<EnumMemberDeclarationSyntax>() != null
-            // It's ok to use magic numbers in the GetHashCode method. Note that I am only checking the method name of the sake of simplicity
             || literalExpression.FirstAncestorOrSelf<MethodDeclarationSyntax>()?.Identifier.ValueText == nameof(object.GetHashCode)
-            // It's ok to use magic numbers in pragma directives
             || literalExpression.FirstAncestorOrSelf<PragmaWarningDirectiveTriviaSyntax>() != null
-            // It's ok to use magic numbers in property declaration
             || IsInsideProperty(literalExpression)
-            || IsNamedArgument(literalExpression)
-            || IsSingleOrNamedAttributeArgument(literalExpression)
-            || IsSingleDigitInToleratedComparisons(literalExpression);
+            || IsSingleDigitInToleratedComparisons(literalExpression)
+            || IsToleratedArgument(literalExpression);
+
+        // Inside property we consider magic numbers as exceptions in the following cases:
+        //   - A {get; set;} = MAGIC_NUMBER
+        //   - A { get { return MAGIC_NUMBER; } }
+        private static bool IsInsideProperty(SyntaxNode node)
```

Comment:
this was just reordered from below, no functional change (see in overall changes this part didn't change)